plugins/azure-devops-backend/src/api/AzureDevOpsApi.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from 'winston';
+import { WebApi } from 'azure-devops-node-api';
+import { RepoBuild } from './types';
+
+export class AzureDevOpsApi {
+  constructor(
+    private readonly logger: Logger,
+    private readonly webApi: WebApi,
+  ) {}
+
+  async getGitRepository(projectName: string, repoName: string) {
+    if (this.logger) {
+      this.logger.debug(
+        `Calling Azure DevOps REST API, getting Repository ${repoName} for Project ${projectName}`,
+      );
+    }
+
+    const client = await this.webApi.getGitApi();
+    return client.getRepository(repoName, projectName);
+  }
+
+  async getBuildList(projectName: string, repoId: string, top: string) {
```

Comment:
i'll suggest that the router should be responsible for the unpacking the request concerns; this inner method should have `top: number` because it makes sense to itself, and be able to trust that the caller gives it the correct thing.

(same below)

Reply:
Makes sense to me, I've made this change